src/Pim/Bundle/CatalogBundle/Elasticsearch/Sorter/Attributes/AbstractAttributeSorter.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Pim\Bundle\CatalogBundle\Elasticsearch\Sorter\Attributes;
+
+use Pim\Bundle\CatalogBundle\Elasticsearch\SearchQueryBuilder;
+use Pim\Component\Catalog\Exception\InvalidDirectionException;
+use Pim\Component\Catalog\Model\AttributeInterface;
+use Pim\Component\Catalog\Query\Sorter\AttributeSorterInterface;
+use Pim\Component\Catalog\Query\Sorter\Directions;
+
+/**
+ * Attribute base sorter for an Elasticsearch query
+ *
+ * @author    Samir Boulil <samir.boulil@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+abstract class AbstractAttributeSorter implements AttributeSorterInterface
+{
+    /** @var SearchQueryBuilder */
+    protected $searchQueryBuilder;
+
+    /** @var array */
+    protected $supportedAttributeTypes;
+
+    /**
+     * @param array $supportedAttributeTypeCodes
+     */
+    public function __construct(array $supportedAttributeTypeCodes = [])
+    {
+        $this->supportedAttributeTypes = $supportedAttributeTypeCodes;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addAttributeSorter(AttributeInterface $attribute, $direction, $locale = null, $scope = null)
+    {
+        if (null === $this->searchQueryBuilder) {
+            throw new \LogicException('The search query builder is not initialized in the sorter.');
+        }
+
+        $attributePath = $this->getAttributePath($attribute, $locale, $scope);
+
+        $suffix = $this->getAttributePathSuffix();
+        if (null !== $suffix) {
+            $attributePath .= '.' . $suffix;
+        }
+
+        switch ($direction) {
+            case Directions::ASCENDING:
+                $sortClause = [
+                    $attributePath => [
+                        'order'   => 'ASC',
+                        'missing' => '_last',
+                    ],
+                ];
+                $this->searchQueryBuilder->addSort($sortClause);
+
+                break;
+            case Directions::DESCENDING:
+                $sortClause = [
+                    $attributePath => [
+                        'order'   => 'DESC',
```

Comment:
No I did this on purpose. I don't want our interface DIRECTIONS's value to be dependant on ES 'DESC' operator